### ISSUE TITLE
Equivalent of getFrameFormat/getVideoFrameFormat

### ISSUE THREAD

**Issue by Asd-g** [AviSynth/AviSynthPlus#240]
Hi, I cannot find an equivalent of [getFrameFormat/getVideoFrameFormat](https://www.vapoursynth.com/doc/api/vapoursynth.h.html?highlight=getframeformat#getframeformat). Any chance to be added?

**Comment by pinterf** [comment]
I don't think so. class VideoFrame is quite a low-level structure. For example you don't even guess the real width only the absolute byte-width (row_size). VideoInfo cannot be reconstructed from that.

**Comment by Asd-g** [comment]
Any suggestion how to retrieve the format of frame from env->propGetFrame() ?

**Comment by pinterf** [comment]
So this is the use case. I see. Somehow the actual VideoInfo fields must be stored and restored manually. You can save the actual VideoInfo* data with sizeof(VideoInfo) length (storing a byte array in frame props is the same as storing a string - both is a byte buffer). If the actual size of VideoInfo would ever be changed by adding fields then it would probably cause troubles.

**Comment by Asd-g** [comment]
Thanks.
